.storybook/webpack.config.js

```diff
@@ -6,34 +6,61 @@
 const path = require('path')
 const webpack = require('webpack')
 const fs = require('fs')
+const config = require('../build/commands/lib/config')
+const genTsConfig = require('../build/commands/lib/genTsConfig')
 const {
   fallback,
   provideNodeGlobals
 } = require('../components/webpack/polyfill')
-
-const buildConfigs = ['Component', 'Static', 'Debug', 'Release']
-const extraArchitectures = ['arm64', 'x86']
-
-function getBuildOuptutPathList(buildOutputRelativePath) {
+const generatePathMap = require('../components/webpack/path-map')
+
+// Choose which brave-core build directory to look for pre-compiled
+// resource dependencies:
+// 1. Default for local builds for the actual platform / architecture
+// 2. platform / architecture overriden by environment variables
+// 3. most recently built - this caters to the common scenario when a
+// non-standard target has been built but no arguments are provided to storybook
+config.update({
+  target_arch: process.env.TARGET_ARCH,
+  target_os: process.env.TARGET_OS,
+  target_environment: process.env.TARGET_ENVIRONMENT,
+  target: process.env.TARGET
+})
+
+let outputPath = config.outputDir
+
+function getBuildOutputPathList() {
   return buildConfigs.flatMap((config) => [
-    path.resolve(__dirname, `../../out/${config}/${buildOutputRelativePath}`),
+    path.resolve(__dirname, `../../out/${config}`),
     ...extraArchitectures.map((arch) =>
       path.resolve(
         __dirname,
-        `../../out/${config}_${arch}/${buildOutputRelativePath}`
+        `../../out/${config}_${arch}`
       )
     )
   ])
 }
 
-const genFolder = getBuildOuptutPathList('gen')
-  .filter((a) => fs.existsSync(a))
-  .sort((a, b) => fs.statSync(b).mtime - fs.statSync(a).mtime)[0]
-if (!genFolder) {
-  throw new Error('Failed to find build output folder!')
+if (fs.existsSync(outputPath)) {
+  console.log('Assuming precompiled dependencies can be found at the existing path found from brave-core configuration: ' + outputPath)
+} else {
+  const outDirectories = getBuildOutputPathList()
+    .filter(a => fs.existsSync(a))
+    .sort((a, b) => fs.statSync(b).mtime - fs.statSync(a).mtime)
```

Comment:
@bridiver this is the section of code I disagree with; I believe its goal is to pick the last-modified `out/` directory (`out/Component`, `out/Release`, etc.) as the directory to use for the build.

Reply:
I'm maintaining current functionality and adding the new functionality on top for now. We can remove in a follow-up.